src/server/pps/server/determined.go

```diff
@@ -0,0 +1,320 @@
+	}
+	// TODO: if user already exists, do we want to still run role assignment for them?
+	// this would involve a querying for the user's id
+	userId, err := a.provisionDeterminedPipelineUser(ctx, pi, tok)
+	if err != nil {
+		return err
+	}
+	roleId, err := a.workspaceEditorRoleId(ctx, tok)
+	if err != nil {
+		return err
+	}
+	if err := a.assignDeterminedPipelineRole(ctx, tok, userId, roleId, detWorkspaces); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *apiServer) workspaceEditorRoleId(ctx context.Context, token string) (int, error) {
+	data, err := a.determinedHttpRequest(ctx, "POST", "/api/v1/roles/search", token, map[string]interface{}{
+		"offset": 0,
+		"limit":  0,
+	})
+	if err != nil {
+		return 0, errors.Wrap(err, "search determined roles")
+	}
+	var roleResponse detSearchRoleResponse
+	if err := json.Unmarshal(data, &roleResponse); err != nil {
+		return 0, errors.Wrap(err, "unmarshal search determined roles response")
+	}
+	for _, r := range roleResponse.Roles {
+		if r.Name == "Editor" {
+			return r.RoleId, nil
+		}
+	}
+	return 0, errors.Errorf("workspaceEditor role not found")
+}
+
+func (a *apiServer) resolveDeterminedWorkspaces(ctx context.Context, token string, workspaces []string) ([]detWorkspace, error) {
+	data, err := a.determinedHttpRequest(ctx, "GET", "/api/v1/workspaces", token, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "list determined workspaces")
+	}
+	var detWorkspaces []detWorkspace
+	if err := json.Unmarshal(data, &detWorkspaces); err != nil {
+		return nil, errors.Wrap(err, "unmarshal determined workspaces")
+	}
+	workspaceSet := make(map[string]struct{})
+	for _, w := range workspaces {
+		workspaceSet[w] = struct{}{}
+	}
+	res := make([]detWorkspace, len(workspaces))
+	for _, dw := range detWorkspaces {
+		if _, ok := workspaceSet[dw.Name]; ok {
+			res = append(res, dw)
+			delete(workspaceSet, dw.Name)
+		}
+	}
+	if len(workspaceSet) > 0 {
+		var errWorkspaces []string
+		for w := range workspaceSet {
+			errWorkspaces = append(errWorkspaces, w)
+		}
+		return nil, errors.Errorf("requested workspaces not found: %v", errWorkspaces)
+	}
+	return res, nil
+}
+
+func (a *apiServer) mintDeterminedToken(ctx context.Context) (string, error) {
+	u, p, err := a.determinedCredentials(ctx)
+	if err != nil {
+		return "", err
+	}
+	m := map[string]interface{}{
+		"username": u,
+		"password": p,
+	}
+	data, err := a.determinedHttpRequest(ctx, "POST", "/api/v1/auth/login", "", m)
+	if err != nil {
+		return "", errors.Wrap(err, "login as determined user")
+	}
+	var respObj detLoginResponse
+	if err := json.Unmarshal(data, &respObj); err != nil {
+		return "", errors.Wrap(err, "unmarshal determined token")
+	}
+	return respObj.Token, nil
+}
+
+func (a *apiServer) determinedCredentials(ctx context.Context) (string, string, error) {
+	secretName := a.env.Config.DeterminedCredentialsSecret
+	secret, err := a.env.KubeClient.CoreV1().Secrets(a.namespace).Get(ctx, secretName, metav1.GetOptions{})
```

Comment:
I think we should avoid the k8s API for this, and inject this secret as either environment variables or as a file (if we want changes to take effect at runtime without having to restart pachd).